admin_guide/scheduling/scheduler.adoc

```diff
@@ -97,185 +356,338 @@ and a string match with the name of the host.
 and all subsequent pods of the service are scheduled on nodes that have the
-same label values.
+same label values as that node.
+
+[source,json]
+----
+"predicates":[
+      {
+         "name":"<name>", <1>
+         "weight" : "1" <2>
+         "argument":{
+            "serviceAffinity":{
+               "labels":[
+                  "<label>" <3>
+               ]
+            }
+         }
+      }
+   ],
+----
+<1> Specify a name for the predicate.
+<2> Specify a weight from 1 (bad fit) to 10 (best fit). 
+<3> Specify a label for matching.
+
+For example:
+
+[source,json]
 ----
-{"name" : "Zone", "argument" : {"serviceAffinity" : {"labels" : ["zone"]}}}
+         "name":"ZoneAffinity",
+         "weight" : "1"
+         "argument":{
+            "serviceAffinity":{
+               "labels":[
+                  "rack"
 ----
 
+//https://github.com/kubernetes/kubernetes/commit/3a383e506a2a447515bbf47895005188041153a0
+For example. if the first pod of a service had a node selector `rack` was scheduled to a node with label `region=rack`, 
+all the other subsequent pods belonging to the same service will be scheduled on nodes 
+with the same `region=rack` label. For more information,
+see xref:../../admin_guide/scheduling/pod_placement.adoc#controlling-pod-placement[Controlling Pod Placement].
+
+Multiple-level labels are also supported. Users can also specify all pods for a service to
+be scheduled on nodes within the same region and within the same zone (under the region).
+
 **_LabelsPresence_** checks whether a particular node has a certain label
 defined or not, regardless of its value. Matching by label can be useful, for
 example, where nodes have their physical location or status defined by labels.
-----
-{"name" : "RequireRegion", "argument" : {"labelsPresence" : {"labels" : ["region"], "presence" : true}}}
-----
-* If "presence" is false, and any of the requested labels match any of the
-node labels, it returns false. Otherwise, it returns true.
-* If "presence" is true, and any of the requested labels do not match any of
+
+[source,json]
+----
+"predicates":[
+      {
+         "name":"<name>", <1>
+         "weight" : "1" <2>
+         "argument":{
+            "labelsPresence":{
+               "labels":[
+                  "<label>" <3>
+               ]
+            }
+         }
+      }
+   ],
+----
+<1> Specify a name for the predicate.
+<2> Specify a weight from 1 (bad fit) to 10 (best fit). 
+<3> Specify a label for matching.
+
+For example:
+
+[source,json]
+----
+         "name":"RackPreferred",
+         "weight" : "1"
+         "argument":{
+            "labelsPresence":{
+               "labels":[
+                  "rack"
+----
+
+* If `presence` is false, and any of the requested labels match any of the
+node labels, it returns `false`. Otherwise, it returns `true`.
+* If `presence` is `true, and any of the requested labels do not match any of
 the node's labels, it returns false. Otherwise, it returns true.
```

Comment:
Here false and true should be consistent with above items.

Reply:
@wjiangjay @ravisantoshgudimetla 
What does this mean, _If `presence` is false_ and _If `presence` is true_? Does this mean if the labelsPresence parameter is present in the JSON, this is _true_ and if the parameter is absent, this is _false_?